src/pages/home/search/SearchModal.tsx

```diff
@@ -32,12 +50,38 @@ function SearchModal() {
     }
   };
 
-  const onSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const recentSearch = [...savedSearch];
-    // currently it is saving each word is typed, need to handle with debounce
-    recentSearch.push(event.target.value);
-    setSavedSearch(recentSearch);
+  const onInputValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    setCurrHoveredOption(undefined);
+  };
+
+  const onSearch = (newSearch: string, doNavigate = true) => {
+    setSavedSearch((prev) => {
+      const updatedSearch = [...prev, newSearch];
+      const uniqueSearch = [...new Set(updatedSearch)];
+      return uniqueSearch;
```

Comment:
need to limit the search to latest 5 , to save the localstorage

Reply:
Saved prev 50 search in local storage and show only 5 prev search at max in autocomplete option list